### PR TITLE
Update internal db storage

### PR DESCRIPTION
**Pull request by kremnik**
### What has been done

With this PR, the performance of the `in` and `update` operations has been improved by replacing `list` with `set` in the `recognition` module.

## How to test
```shell
make lint && make test
```

### CODE REVIEW

File: deepface/modules/recognition.py [hunk 1]
```diff
@@ -194,8 +192,8 @@ def find(
         )
 
     # append replaced images into both old and new images. these will be dropped and re-added.
-    new_images = new_images + replaced_images
-    old_images = old_images + replaced_images
+    new_images.update(replaced_images)
```
Comment (serengil) [inline deepface/modules/recognition.py:195]: suppose that `refresh_database  = False`, new_images and old_images are not initialized anymore in line 163 and 164. Line 195 and 196 will throw expception in that case.

Additionally, we should have an unit test for refresh_database = False

**Comment by serengil** [comment]
Secondly, do we have any stats about the improvement? The question is that will it be worth?

**Comment by kremnik** [comment]
@serengil There are three reasons: 
1. Converting `set` to `list` is time consumption, but not neccessary operation;
2. There are a few places when membership checking occurs: [here](https://github.com/serengil/deepface/blob/ff03dc6ece5bc5b38536228766d94b873087d4ed/deepface/modules/recognition.py#L179C1-L182C25) and [here](https://github.com/serengil/deepface/blob/ff03dc6ece5bc5b38536228766d94b873087d4ed/deepface/modules/recognition.py#L200C1-L203C32). If `old_images` includes a huge amount of images (`n`), as well as `representations` includes another huge amount of images (`m`), total time complexity will be `O(nm)` if `old_images` is a `list`. Instead, if `old_images` is a `set`, total time complexity will be `O(m)`;
3. It is guaranteed that there is no re-processing of the same images.

**Comment by serengil** [comment]
LGTM

**Pull request merged**